sim/src/bookinfo/productpage.rs

```diff
@@ -11,168 +12,88 @@ use std::cmp::min;
 use std::fmt;
 
 pub struct ProductPage {
-    queue: Queue<Rpc>,             // queue of rpcs
-    id: String,                    // id of the node
-    capacity: u32,                 // capacity of the node;  how much it can hold at once
-    egress_rate: u32,              // rate at which the node can send out rpcs
-    generation_rate: u32, // rate at which the node can generate rpcs, which are generated regardless of input to the node
-    plugin: Option<PluginWrapper>, // filter to the node
-    neighbors: Vec<String>, // who is the node connected to
-    seed: u64,
+    core_node: Node,
 }
 
 impl fmt::Display for ProductPage {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if let Some(width) = f.width() {
-            if self.plugin.is_none() {
-                write!(f, "{:width$}",
-                       &format!("ProductPage {{ id : {}, capacity : {}, egress_rate : {}, generation_rate : {}, queue: {}, plugin : None}}",
-                       &self.id, &self.capacity, &self.egress_rate, &self.generation_rate, &self.queue.size()),
-                       width = width)
-            } else {
-                write!(f, "{:width$}",
-                       &format!("ProductPage {{ id : {}, capacity : {}, egress_rate : {}, generation_rate : {}, queue : {}, \n\tplugin : {} }}",
-                       &self.id, &self.capacity, &self.egress_rate, &self.generation_rate, &self.queue.size(), self.plugin.as_ref().unwrap()),
-                       width = width)
-            }
-        } else {
-            if self.plugin.is_none() {
-                write!(f, "ProductPage {{ id : {}, egress_rate : {}, generation_rate : {}, plugin : None, capacity : {}, queue : {} }}",
-                       &self.id, &self.egress_rate, &self.generation_rate, &self.capacity, &self.queue.size())
-            } else {
-                write!(
-                    f,
-                    "ProductPage {{ id : {}, egress_rate : {}, generation_rate : {}, plugin : {}, capacity : {}, queue : {} }}",
-                    &self.id,
-                    &self.egress_rate,
-                    &self.generation_rate,
-                    self.plugin.as_ref().unwrap(),
-                    &self.capacity,
-                    &self.queue.size()
-                )
-            }
-        }
+        node_fmt_with_name(&self.core_node, f, "ProductPage")
     }
 }
 
 impl SimElement for ProductPage {
     fn tick(&mut self, tick: u64) -> Vec<(Rpc, String)> {
         let mut ret = vec![];
+        let review_nodes = vec!["reviews-v1", "reviews-v2", "reviews-v3"];
+        let mut rng: StdRng = SeedableRng::seed_from_u64(self.core_node.seed);
+
         for _ in 0..min(
-            self.queue.size() + (self.generation_rate as usize),
-            self.egress_rate as usize,
+            self.core_node.queue.size(),
```

Comment:
This sould still be queue size plus generation rate.  We should allow productpage, reviews, etc to create multiple child spans, in order to support branching trees.